relay/worker/execution.go

```diff
@@ -37,18 +40,21 @@ func ExecutionWorker(queue chan interface{}) {
 			log.Error("Dropping improperly queued request.")
 			continue
 		}
-
 		invoke := ctx.Value("invoke").(*CommandInvocation)
-		executeCommand(invoke)
+		if bufferedReader == nil {
+			bufferedReader = bufio.NewReader(bytes.NewReader(invoke.Payload))
+			decoder = json.NewDecoder(bufferedReader)
+		} else {
+			bufferedReader.Reset(bytes.NewReader(invoke.Payload))
+		}
+		executeCommand(decoder, invoke)
 	}
 }
 
-func executeCommand(invoke *CommandInvocation) {
+func executeCommand(decoder *json.Decoder, invoke *CommandInvocation) {
 	request := &messages.ExecutionRequest{}
 
-	d := json.NewDecoder(bytes.NewReader(invoke.Payload))
-	d.UseNumber()
```

Comment:
Is this going to take us back to treating integers > 1000000 like floats in scientific notation?